suite-common/fiat-services/src/coingecko.ts

```diff
@@ -103,39 +103,57 @@ export const fetchCurrentFiatRates = async (ticker: TickerId) => {
 };
 
 /**
- * Returns the historical rate for a given symbol, timestamp fetched from CoinGecko API.
- * Be aware that the data granularity is 1 day.
- * Returns null if coin for a given symbol was not found.
+ * Helper function that goes through timestamped fiat rates returned from Coingecko and finds the closest one to the provided timestamp.
+ * @returns [timestamp, fiatRate] pair
+ */
+const findClosestTimestampValue = (timestamp: number, arr: Array<[number, number]>): number => {
+    let closestTimestamp = arr[0];
+    for (let i = 1; i < arr.length; i++) {
+        const currentTimeDelta = Math.abs(timestamp - closestTimestamp[0] / 1000);
+        const nextTimeDelta = Math.abs(timestamp - arr[i][0] / 1000);
+
+        closestTimestamp = arr[i];
+
+        // The timestamps are ordered, if next time delta is higher, we can stop the iteration.
+        if (currentTimeDelta < nextTimeDelta) {
+            break;
+        }
+    }
+
+    return closestTimestamp[1];
```

Comment:
I'm getting this error when I try to load ADA account (allall seed) in web version.
```
TypeError: Cannot read properties of undefined (reading '1')
    at findClosestTimestampValue (coingecko.ts:123:28)
```
    
![image](https://github.com/trezor/trezor-suite/assets/3729633/1d298998-2e48-4f69-93b5-b4ace9672b6b)

Reply:
empty `arr` has to be handled, but I believe the problem is in `ensureHistoryRates` method, that empty `missingRates` array is not handled.

Reply:
Good catch. I will look at it.

Reply:
Handling of empty arrays added in 6d30b1734

Reply:
Fixed in https://github.com/trezor/trezor-suite/commit/6d30b17342d54944eb60074bb84409a98ad0c035.